src/goto-programs/restrict_function_pointers.cpp

```diff
@@ -0,0 +1,415 @@
+/*******************************************************************\
```

Comment:
This file is pretty big - I'd be tempted to split up the command line option parsing / error checking from the goto code transformation stuff

Reply:
450 lines is still within reasonable bounds IMO